azure/04-cluster-services/index.ts

```diff
@@ -0,0 +1,21 @@
+import * as azure from "@pulumi/azure";
+import * as pulumi from "@pulumi/pulumi";
+import { config } from "./config";
+
+const name = pulumi.getProject();
+
+// Enable the Monitoring Diagonostic control plane component logs and AllMetrics
+const azMonitoringDiagnostic = new azure.monitoring.DiagnosticSetting(name, {
+    logAnalyticsWorkspaceId: config.logAnalyticsWorkspaceId,
+    targetResourceId: config.clusterId,
+    logs: [{
+            category: "kube-apiserver",
+            enabled : true,
+            retentionPolicy: { enabled: true },
+        },
+    ],
+    metrics: [{
+        category: "AllMetrics",
+        retentionPolicy: { enabled: true },
+    }],
+});
```

Comment:
It may be worth it to add a note about worker / kubelet logs in AKS requiring users SSH into the nodes directly: https://docs.microsoft.com/en-us/azure/aks/kubelet-logs - this may be best left for the docs portion of this stack since we have no code to add